From cbb495334ce6205251d47a394f2a5154d0963ed5 Mon Sep 17 00:00:00 2001
From: Gabe Black <gabe.black@gmail.com>
Date: Sat, 11 Sep 2021 00:03:52 -0700
Subject: [PATCH 329/757] cpu: Make getIsaPtr const.

This accessor shouldn't modify the ThreadContext.

Change-Id: I6f2629f950b0379573e51e337626ef6b1429a14d
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/50250
Reviewed-by: Giacomo Travaglini <giacomo.travaglini@arm.com>
Maintainer: Giacomo Travaglini <giacomo.travaglini@arm.com>
Tested-by: kokoro <noreply+kokoro@google.com>
---
 src/arch/arm/fastmodel/iris/thread_context.hh | 2 +-
 src/cpu/checker/thread_context.hh             | 2 +-
 src/cpu/o3/thread_context.hh                  | 2 +-
 src/cpu/simple_thread.hh                      | 2 +-
 src/cpu/thread_context.hh                     | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/arch/arm/fastmodel/iris/thread_context.hh b/src/arch/arm/fastmodel/iris/thread_context.hh
index 6464019..2feb25e 100644
--- a/src/arch/arm/fastmodel/iris/thread_context.hh
+++ b/src/arch/arm/fastmodel/iris/thread_context.hh
@@ -220,7 +220,7 @@ class ThreadContext : public gem5::ThreadContext
     System *getSystemPtr() override { return _cpu->system; }
 
     BaseISA *
-    getIsaPtr() override
+    getIsaPtr() const override
     {
         return _isa;
     }
diff --git a/src/cpu/checker/thread_context.hh b/src/cpu/checker/thread_context.hh
index acff791..13828f5 100644
--- a/src/cpu/checker/thread_context.hh
+++ b/src/cpu/checker/thread_context.hh
@@ -153,7 +153,7 @@ class CheckerThreadContext : public ThreadContext
         return checkerCPU;
     }
 
-    BaseISA *getIsaPtr() override { return actualTC->getIsaPtr(); }
+    BaseISA *getIsaPtr() const override { return actualTC->getIsaPtr(); }
 
     InstDecoder *
     getDecoderPtr() override
diff --git a/src/cpu/o3/thread_context.hh b/src/cpu/o3/thread_context.hh
index 9135e1d..ae37dc9 100644
--- a/src/cpu/o3/thread_context.hh
+++ b/src/cpu/o3/thread_context.hh
@@ -107,7 +107,7 @@ class ThreadContext : public gem5::ThreadContext
     CheckerCPU *getCheckerCpuPtr() override { return NULL; }
 
     BaseISA *
-    getIsaPtr() override
+    getIsaPtr() const override
     {
         return cpu->isa[thread->threadId()];
     }
diff --git a/src/cpu/simple_thread.hh b/src/cpu/simple_thread.hh
index c2e620c..2f5805e 100644
--- a/src/cpu/simple_thread.hh
+++ b/src/cpu/simple_thread.hh
@@ -208,7 +208,7 @@ class SimpleThread : public ThreadState, public ThreadContext
 
     CheckerCPU *getCheckerCpuPtr() override { return NULL; }
 
-    BaseISA *getIsaPtr() override { return isa; }
+    BaseISA *getIsaPtr() const override { return isa; }
 
     InstDecoder *getDecoderPtr() override { return decoder; }
 
diff --git a/src/cpu/thread_context.hh b/src/cpu/thread_context.hh
index 093d48c..835ac46 100644
--- a/src/cpu/thread_context.hh
+++ b/src/cpu/thread_context.hh
@@ -142,7 +142,7 @@ class ThreadContext : public PCEventScope
 
     virtual CheckerCPU *getCheckerCpuPtr() = 0;
 
-    virtual BaseISA *getIsaPtr() = 0;
+    virtual BaseISA *getIsaPtr() const = 0;
 
     virtual InstDecoder *getDecoderPtr() = 0;
 
-- 
1.8.3.1

